core/src/avm1/value.rs

```diff
@@ -149,6 +149,11 @@ impl<'gc> Value<'gc> {
         let v = match self {
             Value::Undefined if activation.swf_version() < 7 => return 0.0,
             Value::Null if activation.swf_version() < 7 => return 0.0,
+            Value::Object(_) if activation.swf_version() < 5 => return 0.0,
+            Value::String(v) if activation.swf_version() < 5 => {
+                use crate::avm1::globals::parse_float_impl;
+                return parse_float_impl(v.trim_start(), true, 0.0);
+            }
```

Comment:
Instead of adding the `default_return_value` param to `parse_float`, could this case instead check if the return value of `parse_float_impl(val)` and return 0 in that case? That way the change stays local to only this function.